File: returned/python/today4/merge_intervals.py
# https://leetcode.com/problems/merge-intervals/description/
from typing import List

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        result: List[List[int]] = []
        prev_min = intervals[0][0]
        prev_max = intervals[0][1]
        for interval in intervals:
            if prev_max >= interval[0] and prev_min <= interval[1]:
                prev_min = min(prev_min, interval[0])
                prev_max = max(prev_max, interval[1])
            else:
                result.append([prev_min, prev_max])
                prev_min = interval[0]
                prev_max = interval[1]
        result.append([prev_min, prev_max])
        return result

File: csharp/Leet/Today2/MergeIntervals.cs
// https://leetcode.com/problems/merge-intervals/description/
namespace Today2.MergeIntervals;

public class Solution
{
    public int[][] Merge(int[][] intervals)
    {
        Array.Sort(intervals, (a, b) => a[0].CompareTo(b[0]));
        var result = new List<int[]>();
        var i = 0;
        while (i < intervals.Length)
        {
            var largest = intervals[i][1];
            var j = i + 1;
            while (j < intervals.Length && largest >= intervals[j][0])
            {
                largest = Math.Max(largest, intervals[j][1]);
                j++;
            }
            result.Add([intervals[i][0], largest]);
            i = j;
        }
        return result.ToArray();
    }
}